webpack/localization-plugin/src/LocalizationPlugin.ts

```diff
@@ -157,43 +159,51 @@ export class LocalizationPlugin implements Webpack.Plugin {
               return;
             }
 
-            const chunkFiles: string[] = chunkGroup.getFiles();
-            for (const chunkFileName of chunkFiles) {
-              if (
-                chunkFileName.match(Constants.LOCALE_FILENAME_PLACEHOLDER_REGEX) && // Ensure this is expected to be localized
-                chunkFileName.endsWith('.js') && // Ensure this is a JS file
-                !alreadyProcessedAssets.has(chunkFileName) // Ensure this isn't a vendor chunk we've already processed
-              ) {
-                alreadyProcessedAssets.add(chunkFileName);
-
-                const asset: IAsset = compilation.assets[chunkFileName];
-                const resultingAssets: Map<string, IProcessAssetResult> = this._processAsset(
-                  compilation,
-                  chunkFileName,
-                  asset
-                );
-
-                // Delete the existing asset because it's been renamed
-                delete compilation.assets[chunkFileName];
-
-                const localizedChunkAssets: { [locale: string]: string } = {};
-                for (const [locale, newAsset] of resultingAssets) {
-                  compilation.assets[newAsset.filename] = newAsset.asset;
-                  localizedChunkAssets[locale] = newAsset.filename;
-                }
+            for (const chunk of chunkGroup.chunks) {
+              // Clone the chunk files array because we're going to modify it
+              const chunkFiles: string[] = [...chunk.files];
+              const chunkFilesSet: Set<string> = new Set(chunkFiles);
+              for (const chunkFileName of chunkFiles) {
```

Comment:
can this be optimized?

Reply:
Not really. It'll still end up iterating through the same number of elements, regardless of how we do the iteration.